### PR TITLE
Feature/configurable function map

### PR DESCRIPTION
**Pull request by timjonesdev**

### CODE REVIEW

**Comment by tremaineeto** [comment]
I was able to pull this in and test that it builds and then works fine in k8s.

**Review by tremaineeto — COMMENTED** [review]
I was able to pull this in and test that it builds and then works fine in k8s.

File: main.go [hunk 1]
```diff
@@ -42,49 +73,39 @@ func main() {
...
-							fmt.Printf("%s / %s has %s restarts, which is over the %s restart limit.", item.Namespace,
-								item.Name, strconv.Itoa(int(status.RestartCount)), strconv.Itoa(int(restartThreshold)))
-						} else if reasonInWaitingReasons {
-							fmt.Printf("%s / %s has a status of %s, which is configured to be deleted.", item.Namespace,
-								item.Name, reason)
-						}
 
-						rs, err := clientset.AppsV1().ReplicaSets(item.Namespace).Get(item.OwnerReferences[0].Name, metav1.GetOptions{})
+					crawlerConfigDetails, ok := waitingReasons[reason]
+					if ok {
```
Comment (tremaineeto) [inline main.go]: Maybe we can consider changing this "ok" to something more descriptive? Or is "ok" good enough?

Comment (timjonesdev) [reply main.go]: maybe 'exists'?

Comment (tremaineeto) [reply main.go]: That would work. Other ones I thought of might be 'included', 'found', 'inside', etc. Probably any of them are fine. Maybe it's just me but 'ok' makes me think 200 OK as in an HTTP response code so interesting it's sort of a Golang pseudo-convention.

Considering it is pretty common in Go code, I'm fine leaving it as is if it's clear enough.

Comment (tremaineeto) [reply main.go]: Looks like Effective Go references it as a the "comma ok" idiom for this purpose, so actually, I think it's best we leave it. For Go-familiar developers, "ok" might actually be the most descriptive.

https://golang.org/doc/effective_go.html#maps

Comment (tremaineeto) [reply main.go]: Also, as @bhaskar-reddy has mentioned in a different PR and that offset() example in Effective Go shows, I think in this case we can make these two lines into:

```
if crawlerConfigDetails, ok := waitingReasons[reason]; ok {

...
}
```

Comment (tremaineeto) [reply main.go]: I made the "ok" one-line change in a commit, so going to "resolve" this conversation. Feel free to un-resolve if there are any objections.

**Review by tremaineeto — APPROVED** [review]

**Comment by tremaineeto** [comment]
Formally approved this PR, seeing if any more reviews come in. @timjonesdev

**Review by gbraxton — APPROVED** [review]
lgtm

File: config.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	"github.com/spf13/viper"
+	"log"
+	"os"
+)
+
+// make this config object available outside
+var ConfigObj = KleanerConfig{}
+
+// This map would need to be updated to account for any other behavior we would want to enable
+// The keys in this map have to map directly to the DeleteFuncString in the CrawlerConfigDetails struct
```
Comment (timjonesdev) [inline config.go:13]: looks good

**Pull request merged**